src/ghdcbot/bot.py

```
@@ -425,21 +430,141 @@ async def profile_cmd(
+                now_utc = datetime.now(timezone.utc)
+                start_30 = now_utc - timedelta(days=30)
+
+                def _fetch_metrics():
+                    metrics_list = get_contribution_metrics(storage, start_30, now_utc)
+                    user_metrics = next((m for m in metrics_list if m.github_user == github_user), None)
+                    ranked_30 = rank_by_activity(metrics_list)
+                    rank = get_rank_for_user(ranked_30, github_user)
+                    return user_metrics, rank
+
+                user_metrics, rank = await asyncio.to_thread(_fetch_metrics)
+            except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)